orchagent/neighorch.cpp

```diff
@@ -15,10 +17,261 @@ extern RouteOrch *gRouteOrch;
+    auto ip_addr = ip.getIp();
+
+    if (ip.isV4())
+    {
+        nd_msg_p->ndm_family = AF_INET;
+        memcpy(RTA_DATA(rta_p), &ip_addr.ip_addr.ipv4_addr, addr_len);
+    }
+    else
+    {
+        nd_msg_p->ndm_family = AF_INET6;
+        memcpy(RTA_DATA(rta_p), &ip_addr.ip_addr.ipv6_addr, addr_len);
+    }
+
+    // Fill in the MAC address
+    auto mac_len = ETHER_ADDR_LEN;
+    auto mac_addr = mac.getMac();
+
+    rta_p = static_cast<struct rtattr *>
+          (nlmsg_reserve(msg_p, sizeof(struct rtattr) + mac_len, NLMSG_ALIGNTO));
+    if (!rta_p)
+    {
+        SWSS_LOG_ERROR("Netlink rtattr (MAC) failed for '%s'", ip.to_string().c_str());
+        nlmsg_free(msg_p);
+        return false;
+    }
+
+    rta_p->rta_type = NDA_LLADDR;
+    rta_p->rta_len = static_cast<short>(RTA_LENGTH(mac_len));
+    memcpy(RTA_DATA(rta_p), mac_addr, mac_len);
+
+    return send_message(m_nl_sock, msg_p);
+}
+/*
+ * Function Name: processFDBUpdate
+ * Description: Goal of this function is to delete neighbor/ARP entries
+ *              when a port belonging to a VLAN gets deleted. 
+ *              This function is called whenever neighbor orchagent receives
+ *              SUBJECT_TYPE_FDB_CHANGE notification. Currently we only care for
+ *              deleted FDB entries. We flush neighbor entry that matches its
+ *              in-coming interface and MAC with FDB entry's VLAN name and MAC
+ *              respectively. Also this ensures that underlying physical port is
+ *              being deleted by checking its init status.
+ * IN parameters: FdbUpdate
+ * Returns: true if successfully flushes any ARP entry, false otherwise.
+ */
+bool NeighOrch::processFDBUpdate(const FdbUpdate& update)
+{
+    MacAddress fdbEntryMac = update.entry.mac;
+
+    if (update.add)
+    {
+        // For now we are interested only in deleted FDB entries
+        return true;
+    }
+
+    SWSS_LOG_NOTICE("Received FDB update, Flushing all ARP entries with matching MAC");
+
+    // Get Vlan object
+    Port vlan;
+    if (!m_portsOrch->getPort(update.entry.bv_id, vlan))
+    {
+        SWSS_LOG_NOTICE("FdbOrch notification: Failed to locate vlan port from bv_id 0x%" PRIx64 ".", update.entry.bv_id);
+        return false;
+    }
+
+    if (update.port.m_admin_state_up || update.port.m_oper_status == SAI_PORT_OPER_STATUS_UP)
+    {
+        
+        SWSS_LOG_NOTICE("port %s is admin UP. Could be an AGED entry. Dont flush ARP.", update.port.m_alias.c_str());
+        return false;
+    }
+
+    // If the FDB entry MAC matches with neighbor/ARP entry MAC,
+    // and ARP entry incoming interface matches with VLAN name,
+    // flush neighbor/arp entry.
+    for (const auto &entry : m_syncdNeighbors)
+    {
+        if (entry.first.alias == vlan.m_alias && entry.second == fdbEntryMac)
+        {
+            return flushNeighborEntry(entry.first, entry.second);
+        }
+    }
+    return true;
+}
+
+bool NeighOrch::processPortUpdate(const PortUpdate& update)
```

Comment:
This is not needed,  in this case, the kernel will remove the neighbor entries.  

We could potentially use this one to flush neighbor entries by  finding the neighbor entries that has the mac on this port.  this could be done in later PR.